hw4/02-sessions.js

```diff
@@ -1,17 +1,38 @@
+
 const express = require('express');
 const session = require('express-session');
-const app = express();
-const port = process.env.PORT || 5001;
+const app     = express();
+const port    = process.env.PORT || 5001;
+
 
-// Add your code here
+app.use(session({secret: 'secret_key'}));
 
-// Use the express-session module
-app.use(/** ... */);
 
 app.get('/', (req, res) => {
-  // Add your code here
+  if (req.session.first_time) {
+    req.session.routes.push('/');
+    res.send(`Navigate to any route.`);
+  }
+  else {
+    req.session.first_time = true;
+    req.session.routes = ['/'];
+    res.send(`<h1>Welcome to http://localhost:${port}</h1>`);
+  }
+  res.end();
+});
+
+
+app.get('/*', (req, res) => {
+  if (!req.session.routes || req.url === '/favicon.ico') res.redirect('/');
```

Comment:
What if the user enter through a different URL like `http://localhost:5001/random` and it would still force the user to go through `/` and it would have ignored to capture the path of the first visit.

Reply:
That's true. I purposefully did this as a design choice, so that the user would have to visit '/' first (to see the welcome message). Now that I think about it, that wasn't necessary.